pyvista/core/filters/data_set.py

```diff
@@ -384,7 +401,17 @@ def _validate_vector(vector, name):
                 vector = _validation.validate_array3(vector, dtype_out=float, name=name)
             return vector
 
-        axes, std = pyvista.principal_axes(self.points, return_std=True)
+        # Get points and compute principal axes
+        input_mesh = self.cell_centers() if cell_centers else self
+        if merge_points:
+            append_filter = _vtk.vtkAppendFilter()
+            append_filter.MergePointsOn()
+            append_filter.AddInputData(input_mesh)
+            append_filter.Update()
+            points = pyvista.wrap(append_filter.GetOutput()).points  # type: ignore[union-attr]
```

Comment:
Once #6551 is merged:
```suggestion
            points = input_mesh.merge_points().points
```